# Instructions for Part 1:

Given the germline variant call (VCF), find 10 genes on the chromosome you are assigned with
the highest mutational burden (i.e., number of mutations). List the genes and submit records of
the variants you identified in the prioritized genes in a file called gene_variants_chr{i}.vcf, where i
is the number of the chromosome your team is assigned. Example variant file name is
gene_variants_chr2.vcf. In your report, describe the steps you take to identify the variants in the
genes of interest. Make sure to mention any database or software tool you use. If you write your
own code, please make sure to include it in the final submission.
[Extra credit] Suggest an alternative approach (besides using the number of point mutations in
each gene) to prioritize 10 genes. These can include methods that rely on genomic mutations
(finding genes with more pathologically relevant mutations) or other information (scoring genes
using information other than variant counts). Please submit preliminary results of your alternative
approach in a supplementary PDF should you decide to work on the extra credit section.

## ANNOVAR Files



## ENSEMBL files:



## Extra Credit Portion:


